extension/landing-zone/main.tf

```diff
@@ -0,0 +1,85 @@
+##############################################################################
+# Locals
+##############################################################################
+
+locals {
+  resource_group_id = var.resource_group != null ? data.ibm_resource_group.existing_resource_group[0].id : ibm_resource_group.resource_group[0].id
+  ssh_key_id        = var.ssh_key != null ? data.ibm_is_ssh_key.existing_ssh_key[0].id : ibm_is_ssh_key.ssh_key[0].id
+}
+
+##############################################################################
+# Resource Group
+# (if var.resource_group is null, create a new RG using var.prefix)
+##############################################################################
+
+resource "ibm_resource_group" "resource_group" {
+  count    = var.resource_group != null ? 0 : 1
+  name     = "${var.prefix}-rg"
+  quota_id = null
+}
+
+data "ibm_resource_group" "existing_resource_group" {
+  count = var.resource_group != null ? 1 : 0
+  name  = var.resource_group
+}
+
+##############################################################################
+# SSH key
+##############################################################################
+resource "tls_private_key" "tls_key" {
+  count     = var.ssh_key != null ? 0 : 1
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
```

Comment:
As per https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/pull/529/files#r1325602924 I dont think this DA extenaion should create keys - it should only support using an existing key, or adding a key to the VPC region